src/xc20pEncryption.ts

```diff
@@ -78,6 +79,84 @@ export function x25519Encrypter(publicKey: Uint8Array, kid?: string): Encrypter
   return { alg, enc: 'XC20P', encrypt, encryptCek }
 }
 
+export type X25519AuthEncryptParams = {
+  kid?: string,
+  skid?: string,
+  apu?: string,
+  apv?: string
+}
+
+export function x25519AuthEncrypter(recipientPublicKey: Uint8Array, senderSecretKey: Uint8Array, 
+  options: Partial<X25519AuthEncryptParams> = {}): Encrypter {
+
+  const alg = 'ECDH-1PU+XC20PKW'
+  const keyLen = 256
+  const crv = 'X25519'
+
+  // It is RECOMMENDED by the ECDH-1PU spec to set apu and apv 
+  // to base64url encoded kid and base64url encoded skid in the
+  // recipient header. For the KDF, if no apu and apv is provided,
+  // it is RECOMMENDED that PartyVInfo and PartyUInfo contain
+  // the base64url decoded skid and kid.
+  function setPartyInfo(partyInfo, fallback): { encoded: string, raw: Uint8Array } {
+    if (typeof partyInfo === 'undefined') {
+      return (typeof fallback !== 'undefined') ? 
+        { encoded: encodeBase64url(fallback), raw: fromString(fallback) } :
+         { encoded: undefined, raw: undefined }
```

Comment:
Plz make this more readable by using another `if` statement.

Reply:
this is no longer needed, removed it entirely